### PR TITLE
Fix travis integration on external PRs

### PR DESCRIPTION
**Pull request by AlecAivazis**
This PR updates the travis config file to avoid running the coverage generation script when the required environment variable isn't set. 

cc @bk-splunk

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,8 +18,9 @@ install:
 script:
   # install the dependencies
   - run install
-  # generate and report the test coverage
-  - run tests:coverage
+  # generate and report the test coverage if we have a token for coveralls
+  - 'if [ "$COVERALLS_TOKEN" != "" ]; then run tests:coverage; fi'
+  - 'if [ "$COVERALLS_TOKEN" = "" ]; then run tests; fi'
```
Comment (bardhi) [inline .travis.yml]: I think this needs to be `==`

Comment (AlecAivazis) [reply .travis.yml]: So I pulled this from [this doc page](https://docs.travis-ci.com/user/pull-requests/) on travis and it does look like the correct test script was run if I look at the travis build. That being said, I'm definitely far from a bash expert (didn't even occur to me to use `-z` or `-n`) so you could definitely be correct.

Comment (AlecAivazis) [reply .travis.yml]: I did some digging and they are both valid equality operators. `=` works with the `test [` command and `==` works inside of `[[` commands

Comment (bardhi) [reply .travis.yml]: Ah you are right, I didn't know that -- definitely not a bash expert either.

**Comment by bardhi** [comment]
LGTM

**Pull request merged**